### ISSUE TITLE
TypeError: t.versions.node is undefined

### ISSUE THREAD

**Issue by alexredder** [Automattic/mongoose#12392]
### Prerequisites

- [X] I have written a descriptive issue title
- [X] I have searched existing issues to ensure the bug has not already been reported

### Mongoose version

6.6.0

### Node.js version

17.6.0

### MongoDB server version

where do i find that?

### Description

Hey guys,

one day i open my application and suddenly only the following error message pops up:

![grafik](https://user-images.githubusercontent.com/67794362/188335161-3e92222a-b982-4b66-ba2c-921cd77b76d6.png)

I have not changed the code except for updating mongoose.. but i could find out what that error meant or how i can solve it..

this is what it says in chrome:
![grafik](https://user-images.githubusercontent.com/67794362/188335198-d0c486ae-b83d-4b87-bdbc-6230381fd076.png)

I am a newbie so i dont know how to solve this issue.. hope you guys can help me out :)

cheers

### Steps to Reproduce

the error message points to 
**node_modules\mongoose\dist\browser.umd.js (220:268)**
line 220 intails the following code:
`e.noop=function(){},e.errorToPOJO=function(t){if(!(t instanceof Error))throw new Error("`error` must be `instanceof Error`.");var e,r={},o=n(Object.getOwnPropertyNames(t));try{for(o.s();!(e=o.n()).done;){var i=e.value;r[i]=t[i]}}catch(t){o.e(t)}finally{o.f()}return r},e.nodeMajorVersion=parseInt(t.versions.node.split(".")[0],10)}).call(this,r(10))},function(t,e,r){"use strict";(function(t){
`
My Mongo_URI is coming from my .env file

This is my connect code:

> `import * as mongoose from 'mongoose';

const connection = {};
const testvar = process.versions.node;

async function connect() {
  if (connection.isConnected) {
    console.log('already connected');
    return;
  }
  if (mongoose.connections.length > 0) {
    connection.isConnected = mongoose.connections[0].readyState;
    if (connection.isConnected === 1) {
      console.log('use previous connection');
      return;
    }
    await mongoose.disconnect();
  }
  const db = await mongoose.connect(process.env.MONGODB_URI);
  console.log('new connection');
  connection.isConnected = db.connections[0].readyState;
}
console.log(testvar);

async function disconnect() {
  if (connection.isConnected) {
    if (process.env.NODE_ENV === 'production') {
      await mongoose.disconnect();
      connection.isConnected = false;
    } else {
      console.log('not disconnected');
    }
  }
}

function convertDocToObj(doc) {
  doc._id = doc._id.toString();
  doc.createdAt = doc.createdAt.toString();
  doc.updatedAt = doc.updatedAt.toString();
  return doc;
}

const db = { connect, disconnect, convertDocToObj };

export default db;
`

*UPDATE*
i have reset the application to a former state.. now it works in dev-environment but not in my vercel deployment..

### Expected Behavior

run smoothly like it did before.. aswell in deployment as development

**Comment by alexredder** [comment]
@vkarpov15 can you help me here? i have updated to mongoose 6.6.0 but nothing changed so far..

**Comment by lapoguidi** [comment]
Hello,
I have the same problem using mongoose with Sveltekit. I tried also to create an empty Sveltekit skeleton application and after mongoose is added and imported in the browser just to read a Model, the same error appears.

**Comment by alexredder** [comment]
> Hello, I have the same problem using mongoose with Sveltekit. I tried also to create an empty Sveltekit skeleton application and after mongoose is added and imported in the browser just to read a Schema, the same error appears.

have you managed something? - i can revert my code but as soon as i update my packages - its all garbage..

**Comment by Uzlopak** [comment]
Can you provide a simple replication repository?

**Comment by lapoguidi** [comment]
Sure: [https://github.com/lapoguidi/sveltekit-mongoose-http-500](https://github.com/lapoguidi/sveltekit-mongoose-http-500)

I followed just the instructions I reported on [README.md](https://github.com/lapoguidi/sveltekit-mongoose-http-500/blob/main/README.md) and after the start is running, the console.log is executed, but just after an HTTP 500 is received and inside the console I can read: **TypeError: Cannot read properties of undefined (reading 'split')**.

This is the code I've got in _node_modules_
![code-example](https://user-images.githubusercontent.com/14073486/189847636-59df4429-a5c5-47f5-a983-64beec9456e4.jpg)

Thank you for the interest.

**Comment by nathanbabcock** [comment]
Same issue here, using `vite@3.1.2`. My only import is `Schema`, and removing it resolves the error.

**Comment by alexredder** [comment]
@KonghaYao Hi there, i did that in my node_modules/mongoose/dist/browser....js and it solves the issue **locally** - but when i push it to vercel it displays the same error.. please checkout [www.sanivio.de](url) to see in production.

**Comment by KonghaYao** [comment]
@alexredder      Oh, You need to move the dist/browser.umd.js file to your project folder, because your project upload and bundle your source code with  mongoose which has the same bug! You can write  an alias redirect your import to the new file path. 😀

**Comment by alexredder** [comment]
@KonghaYao sorry.. i dont understand can you please explain in more detail where to put it? ..

**Comment by KonghaYao** [comment]
@alexredder 1. You can put browser.js to any path in your project ,like /src/lib/browser.js，
2. if you are using bundler, the bundler like vite will recognize "mongoose" package and find node_module/mongoose always, so we can add an alias like 'mongoose': "/src/lib/browser.js"; vite will automatically redirect mongoose project to your browser.js. 
3. But it is an umd file,  I think you will use script to load it , just src = the browser.js path, browser will load it.

**Comment by Uzlopak** [comment]
I dont know. I played around and it seems that svelte has no global window object, so we can not detect if we have a browser or not.

**Comment by lapoguidi** [comment]
This is not a so useful solution, because everytime that the dependencies are rebuilded, it will be overwritten.
I manually edit everytime the code, but a permanent solution would be better.
Untill now I was not able to solve the code generated during build. I'm hoping that somebody more expert than me can understand and solve this problem.

**Comment by lapoguidi** [comment]
Just an update.

I manually downloaded from npmjs registry the mongoose packages starting from latest _v6.6.2_ untill _v6.5.2_ using an url like this: https://registry.npmjs.org/mongoose/-/mongoose-6.6.2.tgz.
After download I opened one tarball at a time and searched inside _dist/browser.umd.js_ for _nodeMajorVersion_ string.
Well, I found an occurrence in the file starting from _v6.5.3_ and this is the first version that give that issue; _v6.5.2_ doesn't contains that string and seems to be immune from this issue.

Then I simply cloned the _v6.6.2_ repository and tried to build the browser.umd.js.
As I mentioned some post ago, I tried with _node v17.9.0_, _npm v8.19.1_ v and _webpack v5.74.0_.
The resulting _browser.umd.js_ doesn't contain the _nodeMajorVersion_ variable and it seems to work fine.

```
git clone https://github.com/Automattic/mongoose.git
npm i
npm run build-browser
```

Just a couple of questions (for curiosity):

- why the _browser.umd.js_ provided inside the package contains that variable assignment and in the _browser.umd.js_ built with _npm run buil-browser_  doesn't?
- why the resulting _browser.umd.js_ just built is really different from the provided one? (export comments and licenses outside in a separate txt file, for example)

Thank for helping me.
